client/kafka/v2/kafka.go

```diff
@@ -65,16 +65,21 @@ type Builder struct {
 	errs    []error
 }
 
-// New initiates the AsyncProducer/SyncProducer builder chain with the default sarama configuration.
+// New initiates the AsyncProducer/SyncProducer builder chain with an idempotent producer Sarama configuration.
 func New(brokers []string) *Builder {
 	var ee []error
 	if validation.IsStringSliceEmpty(brokers) {
 		ee = append(ee, errors.New("brokers are empty or have an empty value"))
 	}
 
+	cfg := sarama.NewConfig()
+	cfg.Net.MaxOpenRequests = 1
+	cfg.Producer.Idempotent = true
```

Comment:
I'm unsure about this parameter. I haven't seen any library or framework that would default using the exactly-once semantic. I recall testing it in the past and it does have an impact.
Personally, I'd be more comfortable if this wasn't the default config. I mean, users can still decide to use it with `WithConfig` but here it sounds a bit hardcore to me.

Reply:
It has been made the default in the upstream official Java client since last Tuesday: https://blogs.apache.org/kafka/entry/what-s-new-in-apache6

Regardless, we discussed this briefly and I am in the process of submitting an updated PR that leaves choice to the user, sorry for not mentioning it here in the PR so far.

> it sounds a bit hardcore to me.

What sounds hardcore to me is that we are using Kafka producers as if idempotency is always enabled, yet idempotency is always disabled :) it is an issue in case of brokers problems.